cloudprem/physical/eks.tf

```diff
@@ -194,7 +194,8 @@ module "eks_cluster" {
   cluster_name                                   = local.identifier
   cluster_version                                = "1.21"
   enable_irsa                                    = true
-  cluster_endpoint_public_access                 = !var.protect_resources
+  # Need public access even when deploying from AWS due to the occasional inability to access private endpoints.
+  cluster_endpoint_public_access                 = true
```

Comment:
This requires a tfsec annotation?

Reply:
No? There isn't one.

Reply:
Hmm. I thought i remember seeing a tfsec annotation against public cluster endpoints? There's definitely a tfsec rule for that... https://aquasecurity.github.io/tfsec/v0.63.1/checks/aws/eks/no-public-cluster-access/

Reply:
The reason why this doesn't get triggered is because tfsec is set to ignore imported modules. This is mostly due to 2 factors:

1. We aren't able to fix any issues tfsec might flag so it requires us to put a giant block of annotations above the module import which seems exactly as effective as not parsing the module code in the first place.
2. Tfsec is not great at parsing modules in the first place because it doesn't really handle dynamic blocks well. It will report on every possible configuration using that module and many modules allow you to deploy an insecure configuration. So even if our implementation of the module is secure, tfsec will still complain that it isn't. 

If you have a better solution or any recommendations I'd love to hear them 👍

Reply:
Thought I did just run it manually on my workstation with including the downloaded modules and it didn't pickup any issues there either so still not sure what's up with that check.
<img width="901" alt="Screen Shot 2022-01-29 at 2 30 39 AM" src="https://user-images.githubusercontent.com/58958522/151652048-011a857c-2eb3-4bca-9b43-fc5c4952b0a2.png">

Reply:
Ok yep. Just double checked and I was wrong. There is a skip notation but it's at the top of the module declaration, not at that line which is why I missed it: https://github.com/Dozuki/CloudPrem-Infra/blob/7952b9b6e56bd2c81f59553acd24430081918200/cloudprem/physical/eks.tf#L184

Reply:
But you see what I mean? I took off the `#tfsec:ignore:aws-eks-encrypt-secrets` notation because I did add secret encryption here: https://github.com/Dozuki/CloudPrem-Infra/blob/7952b9b6e56bd2c81f59553acd24430081918200/cloudprem/physical/eks.tf#L203-L208 but tfsec doesn't handle modules well so it still thinks secrets aren't encrypted due to the module using a dynamic block:

<img width="998" alt="Screen Shot 2022-01-29 at 2 35 36 AM" src="https://user-images.githubusercontent.com/58958522/151652246-c019a3d3-c2c6-41d2-a319-6aa4366ce46a.png">

So that notation has to remain even thought we do have encrypted secrets. TFsec is great but it still has a long way to go with handling modules.

Reply:
Gotcha

Reply:
Just wanted to double check that.

I think the tfsec rule makes sense though. We probably dont generally want control plane access to be public. But it's also probably not a big deal for now. Not a blocker...